src/meta_schedule/schedule_rule/winograd.cc

```diff
@@ -92,6 +124,50 @@ TVM_REGISTER_GLOBAL("meta_schedule.winograd_data_pack.llvm")
       return {sch};
     });
 
+TVM_REGISTER_GLOBAL("meta_schedule.winograd_output.nchw.cuda")
+    .set_body_typed([](Schedule sch, BlockRV output) -> Array<Schedule> {
+      // get loops
+      Array<LoopRV> loops = sch->GetLoops(output);
+      ICHECK_EQ(loops.size(), 4);
+
+      BlockRV OL{nullptr};
+      if (sch->GetConsumers(output).size() > 0) {
+        OL = GetOnlyConsumer(sch, output);
+        sch->SetScope(OL, /*buffer_index=*/0, /*storage_scope=*/"local");
+      }
+
+      // tile
+      BlockRV inverse = GetOnlyProducer(sch, output);
+      Optional<PrimExpr> tile_size =
+          tir::GetAnn<PrimExpr>(sch->GetSRef(output), "winograd_tile_size");
+      ICHECK(tile_size.defined()) << "Winograd tile size is not defined in block annotation!";
+      Array<LoopRV> split0 = sch->Split(loops[2], {NullOpt, tile_size.value()});
+      Array<LoopRV> split1 = sch->Split(loops[3], {NullOpt, tile_size.value()});
+      sch->Reorder({split0[0], split1[0], split0[1], split1[1]});
+
+      // compute_at
+      sch->ComputeAt(inverse, /*loop_rv=*/split1[0],
+                     /*preserve_unit_loops=*/true);
+      if (OL.defined()) {
+        while (sch->GetConsumers(OL).size() > 0) {
+          BlockRV next_OL = GetOnlyConsumer(sch, OL);
```

Comment:
Will consumers be handled automatically by auto_inline rule? If so, this and `SetScope` are not necessary